src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java

```diff
@@ -0,0 +1,147 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class VoteShowTest extends CreatedDefaultVoteRegistry {
+
+	Vote vote;
+	String voteId;
+	String adminKey;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		voteId = adminInfo.voteId;
+		adminKey = adminInfo.adminKey;
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_name_attribute_contains_the_name_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canAddIn_attribute_contains_the_addability_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canAddIn"), vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_creationTime_attribute_contains_the_creation_time_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_choices_attribute_contains_the_choices_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canEndorse_attribute_contains_the_endorsable_of_the_vote() throws ReportedException {
```

Comment:
the canEndorse attribute tells whether voters endorse choices of the vote at the moment

Reply:
Fixed